### PR TITLE
Create course

### PR DESCRIPTION
**Pull request by phunghangbk**

### CODE REVIEW

File: app/views/courses/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  <%= f.text_field :name, class: "form-control" %>
+
+  <%= f.label :description %>
+  <%= f.text_field :description, class: "form-control" %>
+  <div class="row">
+    <div class="col-md-2">
+      <%= f.label :start %>
+    </div>
+    <div class="col-md-10">
+      <%= f.date_select :start, {prompt: {day: t("Day"), month: t("Month"), year: t("Year")}, :order => [:day, :month, :year], :with_css_classes => true}, { :class => "col-md-4 form-control form-control-inline" } %>
```
Comment (leminhtuan2015) [inline app/views/courses/_form.html.erb]: Convention plx

File: app/views/courses/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    </div>
+  </div>
+
+  <div class="row">
+    <div class="col-md-2">
+      <%= f.label :finish %>
+    </div>
+    <div class="col-md-10">
+      <%= f.date_select :start, {prompt: {day: t("Day"), month: t("Month"), year: t("Year")}, :order => [:day, :month, :year], :with_css_classes => true}, { :class => "col-md-4 form-control form-control-inline" } %>
```
Comment (leminhtuan2015) [inline app/views/courses/_form.html.erb]: Convention

File: app/controllers/courses_controller.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class CoursesController < ApplicationController
+  before_action :logged_in_user, only: [:new, :create]
+
+  def new
+    @course = Course.new
+  end
+
+  def create
+    user_course = UserCourse.new
+    @course = Course.new course_params
```
Comment (leminhtuan2015) [inline app/controllers/courses_controller.rb]: Dont need to use @course , just course

File: app/controllers/courses_controller.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+class CoursesController < ApplicationController
+  before_action :logged_in_user, only: [:new, :create]
+
+  def new
+    @course = Course.new
+  end
+
+  def create
+    user_course = UserCourse.new
```
Comment (leminhtuan2015) [inline app/controllers/courses_controller.rb]: Where will you save user_course

File: app/controllers/courses_controller.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+class CoursesController < ApplicationController
+  before_action :logged_in_user, only: [:new, :create]
+
+  def new
+    @course = Course.new
```
Comment (phunghangbk) [inline app/controllers/courses_controller.rb]: Anh oi, @course em con dung o trong views nua a. Neu dung course thi se bi loi

File: app/views/courses/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  <%= f.text_field :name, class: "form-control" %>
+
+  <%= f.label :description %>
+  <%= f.text_field :description, class: "form-control" %>
+  <div class="row">
+    <div class="col-md-2">
+      <%= f.label :start %>
+    </div>
+    <div class="col-md-10">
+      <%= f.date_select :start, {prompt: {day: t("Day"), month: t("Month"),
```
Comment (leminhtuan2015) [inline app/views/courses/_form.html.erb]:
```ruby
<%= f.date_select :start, {prompt: {day: t("Day"), month: t("Month"), year: t("Year")},
  order: [:day, :month, :year], with_css_classes: true}, {class: "col-md-4 form-control form-control-inline" } %>
```

File: config/routes.rb [hunk 7]
```diff
@@ -6,6 +6,8 @@
   get "login" => "sessions#new"
   post "login" => "sessions#create"
   delete "logout" => "sessions#destroy"
+  get "course_new" => "courses#new"
```
Comment (leminhtuan2015) [inline config/routes.rb]: Tai sao lai can dong nay ha em ey.?

File: app/views/courses/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<% provide :title, t("new_course") %>
+<h1><%= t "new_course" %></h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+      <%= render "form" %>
```
Comment (leminhtuan2015) [inline app/views/courses/new.html.erb]: Convention

**Comment by leminhtuan2015** [comment]
LGTM. well done.

File: app/controllers/application_controller.rb [hunk 9]
```diff
@@ -1,4 +1,11 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   include SessionsHelper
-end
\ No newline at end of file
+
+  def logged_in_user
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: private function

File: app/helpers/courses_helper.rb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+module CoursesHelper
```
Comment (dieunb) [inline app/helpers/courses_helper.rb]: remove this file

File: db/migrate/20150610220828_create_courses.rb [hunk 11]
```diff
@@ -3,7 +3,7 @@ def change
     create_table :courses do |t|
       t.string :name
       t.string :description
-
+
```
Comment (dieunb) [inline db/migrate/20150610220828_create_courses.rb]: remove trailing spaces

**Pull request merged**